### PR TITLE
Add sgs flux contribution to LES fluxes. Simplify parallel.

### PR DESCRIPTION
**Pull request by ilopezgp**
## Changes

- Adds the SGS flux contribution to the PyCLES flux diagnostics for calibration, since the EDMF flux is supposed to represent resolved+unresolved components. This is particularly important near the wall in LES, where the flow is still unresolved even in LES. 

- Adds a descriptive docstring about the differences between certain variables in TC.jl and PyCLES. This docstring also renders in the API section of the CalibrateEDMF documentation for readability.

- Removes a duplicate function in `src/parallel.jl` to improve code robustness.

## Checklist before requesting a review / merging
- [x] I have formatted the code using `julia --project=.dev .dev/climaformat.jl .`
- [x] I have updated all test manifests using `julia --project .dev/up_deps.jl .` with Julia 1.7.3.
- [x] If core features are added, I have added appropriate test coverage.
- [x] If new functions and structs are added, they are documented through docstrings.

### CODE REVIEW

File: integration_tests/SCT1_integration_test_config.jl [hunk 1]
```diff
@@ -25,7 +25,8 @@ namelist_args = [
     ("stats_io", "frequency", 60.0),
     ("stats_io", "calibrate_io", true),
     ("thermodynamics", "sgs", "mean"),
-    ("grid", "nz", 80),
+    ("grid", "stretch", "flag", true),
+    ("grid", "stretch", "dz_surf", 50.0),
```
Comment (yairchn) [inline integration_tests/SCT1_integration_test_config.jl:29]: did you mean to add this stretch in here?

File: test/HelperFuncs/runtests.jl [hunk 2]
```diff
@@ -86,6 +86,8 @@ end
     @test zf[1] ≈ 0
     @test zc[1] > zf[1]
     @test is_face_variable(data_filename, "total_flux_h")
+    @test is_face_variable(data_filename, "total_flux_s")
```
Comment (yairchn) [inline test/HelperFuncs/runtests.jl:89]: above you treat the flux is center variable var_ = rho_half .* (resolved_flux .+ sgs_flux) but here you are testing if it is a face?

Comment (ilopezgp) [reply test/HelperFuncs/runtests.jl:89]: As mentioned in the docstring of `fetch_interpolate_transform`, the fluxes are defined at the faces in TC.jl, but always written at the cell centers in PyCLES. The transforms you see in the function are for LES variables only, the TC variables are never transformed. The comparison is done for an interpolation to `z_scm` of all variables.

File: integration_tests/SCT1_integration_test_config.jl (hunk 1 above)
Comment (ilopezgp) [reply integration_tests/SCT1_integration_test_config.jl:29]: yes! I want to start testing the stretched grid in the integration tests.

**Pull request merged**